### PR TITLE
fix(ddo-train): dataloader epoch seed is never updated (+1 more)

### PR DESCRIPTION
**Pull request by andrewwhitecdw**
Small fixes in `training_code/VAR/DDO_train.py`:

#### fix: dataloader epoch seed is never updated across epochs

**Fix:** Apply patch:
```diff
--- a/training_code/VAR/DDO_train.py
+++ b/training_code/VAR/DDO_train.py
@@ -100,7 +100,6 @@ def build_everything(args: arg_util.Args):
         print(f'[dataloader multi processing] ...', end='', flush=True)
         stt = time.time()
         iters_train = len(ld_train)
-        ld_train = iter(ld_train)
         # noinspection PyArgumentList
         print(f'     [dataloader multi processing](*) finished! ({time.time()-stt:.2f}s)', flush=True, clean=True)
         print(f'[dataloader] gbs={args.glb_batch_size}, lbs={args.batch_size}, iters_train={iters_train}, types(tr, va)={types}')
@@ -198,11 +197,11 @@ def main_training():
     
     L_mean, L_tail = -1, -1
     for ep in range(start_ep, args.ep):
-        if hasattr(ld_train, 'sampler') and hasattr(ld_train.sampler, 'set_epoch'):
-            ld_train.sampler.set_epoch(ep)
+        if hasattr(ld_train, 'batch_sampler') and hasattr(ld_train.batch_sampler, 'set_epoch'):
+            ld_train.batch_sampler.set_epoch(ep)
             if ep < 3:
                 # noinspection PyArgumentList
-                print(f'[{type(ld_train).__name__}] [ld_train.sampler.set_epoch({ep})]', flush=True, force=True)
+                print(f'[{type(ld_train).__name__}] [ld_train.batch_sampler.set_epoch({ep})]', flush=True, force=True)
         tb_lg.set_step(ep * iters_train)
         
         stats, (sec, remain_time, finish_time) = train_one_ep(
-            ep, ep == start_ep, start_it if ep == start_ep else 0, args, tb_lg, ld_train, iters_train, trainer
+            ep, ep == start_ep, start_it if ep == start_ep else 0, args, tb_lg, iter(ld_train), iters_train, trainer
         )
```

#### fix: wrong return type for build_vae_var_with_ref

**Fix:** Replace:
```
) -> Tuple[VQVAE, VAR]:
```

with:
```
) -> Tuple[VQVAE, VAR, VAR]:
```

### Files changed
- `training_code/VAR/DDO_train.py`

### CODE REVIEW

**Comment by andrewwhitecdw** [comment]
Closing this sweep-generated PR: PR has 2 commits; sweep requires exactly one commit per PR. It does not meet the sweep requirements (single signed-off commit).

**Comment by andrewwhitecdw** [comment]
Closing after the adversarial audit found the proposed change was not safe or complete enough for a small focused PR. A revised fix can be submitted separately. Thanks for understanding.

**Comment by andrewwhitecdw** [comment]
Reopening — the previous bulk-close was issued in error. No action needed from maintainers.
